### PR TITLE
[NCL-6661] Implement Error Boundaries

### PR DESCRIPTION
**Pull request by DnsZhou**

### CODE REVIEW

File: src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import SystemErrorPage from '../SystemErrorPage/SystemErrorPage';
+
+export interface ErrorBoundaryProps {
+  children: React.ReactChild;
+  fallbackUI?: React.FunctionComponent;
```
Comment (matedo1) [inline src/components/ErrorBoundary/ErrorBoundary.tsx:6]: I think this line can be removed as it's not implemented in `render()` part

File: src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  meta?: any;
```
Comment (matedo1) [inline src/components/ErrorBoundary/ErrorBoundary.tsx:7]: I think this line can also be removed

File: src/components/ErrorBoundary/__tests__/ErrorBoundary.test.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+import { ErrorBoundary } from '../ErrorBoundary';
+import { render, fireEvent, screen, waitFor } from '@testing-library/react';
+
+test('renders ErrorBoundary and fire an error', async () => {
+  function BuggyCounter() {
+    const [counter, setCounter] = useState<any>(0);
```
Comment (matedo1) [inline src/components/ErrorBoundary/__tests__/ErrorBoundary.test.tsx:7]: `useState<number>`

File: src/components/ErrorBoundary/__tests__/ErrorBoundary.test.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      setCounter(counter + 1);
+    }
+
+    if (counter === 1) {
+      // Simulate a JS error
+      throw new Error('SIMULATE INTENTIONAL CRASH CASES -- This is fine :)');
+    }
+
+    return (
+      <h1 id="error-button" onClick={handleClick}>
```
Comment (matedo1) [inline src/components/ErrorBoundary/__tests__/ErrorBoundary.test.tsx:19]: I would change `<h1>` to `<button>` and then modify also slightly test case to click button with the id `error-button`

**Review by matedo1 — CHANGES_REQUESTED** [review]

File: src/components/ErrorBoundary/__tests__/ErrorBoundary.test.tsx (hunk 4 above)
Comment (DnsZhou) [reply src/components/ErrorBoundary/__tests__/ErrorBoundary.test.tsx:19]: React testing-library provide a property named data-testid, I think this one would be the best for this situation. https://testing-library.com/docs/queries/bytestid

**Review by matedo1 — APPROVED** [review]

**Pull request merged**